flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlTableColumn.java

```diff
@@ -65,34 +67,55 @@ public SqlTableColumn(SqlIdentifier name,
 		this.comment = comment;
 	}
 
+	public SqlTableColumn(SqlIdentifier name,
+			SqlNode expr,
+			@Nullable SqlCharStringLiteral comment,
+			SqlParserPos pos) {
+		super(pos);
+		this.name = requireNonNull(name, "Column name should not be null");
+		this.expr = requireNonNull(expr, "Column expression should not be null");
+		this.comment = comment;
+	}
+
 	@Override
 	public SqlOperator getOperator() {
 		return OPERATOR;
 	}
 
 	@Override
 	public List<SqlNode> getOperandList() {
-		return ImmutableNullableList.of(name, type, comment);
+		return isComputed() ?
+			ImmutableNullableList.of(name, expr, comment) :
+			ImmutableNullableList.of(name, type, comment);
 	}
 
 	@Override
 	public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
 		this.name.unparse(writer, leftPrec, rightPrec);
-		writer.print(" ");
-		this.type.unparse(writer, leftPrec, rightPrec);
-		if (!this.type.getNullable()) {
-			// Default is nullable.
-			writer.keyword("NOT NULL");
-		}
-		if (this.constraint != null) {
-			this.constraint.unparse(writer, leftPrec, rightPrec);
+		if (isComputed()) {
+			writer.keyword("AS");
+			this.expr.unparse(writer, leftPrec, rightPrec);
+		} else {
+			writer.print(" ");
+			this.type.unparse(writer, leftPrec, rightPrec);
+			if (!this.type.getNullable()) {
+				// Default is nullable.
+				writer.keyword("NOT NULL");
+			}
+			if (this.constraint != null) {
+				this.constraint.unparse(writer, leftPrec, rightPrec);
+			}
 		}
 		if (this.comment != null) {
 			writer.print(" COMMENT ");
 			this.comment.unparse(writer, leftPrec, rightPrec);
 		}
 	}
 
+	public boolean isComputed() {
+		return type == null && expr != null;
+	}
```

Comment:
- Can be `isGenerated` to be synced with `TableColumn`
- `return expr != null;` is enough